edge-modules/azure-monitor/src/Program.cs

```diff
@@ -0,0 +1,126 @@
+                if (Settings.Current.UploadTarget == UploadTarget.AzureMonitor) {
+                    publisher = new FixedSetTableUpload.FixedSetTableUpload(Settings.Current.LogAnalyticsWorkspaceId, Settings.Current.LogAnalyticsWorkspaceKey);
+                }
+                else {
+                    publisher = new IotHubMetricsUpload.IotHubMetricsUpload(moduleClient);
+                }
+
+                using (MetricsScrapeAndUpload metricsScrapeAndUpload = new MetricsScrapeAndUpload(scraper, publisher, additionalTags))
+                {
+                    TimeSpan scrapeAndUploadInterval = TimeSpan.FromSeconds(Settings.Current.ScrapeFrequencySecs);
+                    metricsScrapeAndUpload.Start(scrapeAndUploadInterval);
+                    // await cts.Token.WhenCanceled();
+                    WaitHandle.WaitAny(new WaitHandle[] {cts.Token.WaitHandle });
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.Writer.LogError(e, "Error occurred during metrics collection setup.");
+            }
+            finally
+            {
+                moduleClient?.Dispose();
+            }
+
+            completed.Set();
+            handler.ForEach(h => GC.KeepAlive(h));
+
+            Logger.Writer.LogInformation("MetricsCollector Main() finished.");
+            return 0;
+        }
+
+        //TODO: we don't use the module twin for anything, don't bother getting tags?
+        static async Task<Option<SortedDictionary<string, string>>> GetAdditionalTagsFromTwin(ModuleClient moduleClient)
+        {
+            Twin twin = await moduleClient.GetTwinAsync();
+            TwinCollection desiredProperties = twin.Properties.Desired;
+            Logger.Writer.LogInformation($"Received {desiredProperties.Count} tags from module twin's desired properties that will be added to scraped metrics");
+
+            const string additionalTagsPlaceholder = "additionalTags";
+            Dictionary<string, string> deserializedTwin = JsonConvert.DeserializeObject<Dictionary<string, string>>(twin.Properties.Desired.ToJson());
+            if (deserializedTwin.ContainsKey(additionalTagsPlaceholder))
+            {
+                return Option.Some<SortedDictionary<string, string>>(ParseKeyValuePairs(deserializedTwin[additionalTagsPlaceholder], Logger.Writer, true));
+            }
+
+            return Option.None<SortedDictionary<string, string>>();
+        }
+
+        // Test info is used in the longhaul, stress, and connectivity tests to provide contextual information when reporting.
+        // This info is passed from the vsts pipeline and needs to be parsed by the test modules.
+        // Includes information such as build numbers, ids, host platform, etc.
+        // Argument should be in the format key=value[,key=value]
+        public static SortedDictionary<string, string> ParseKeyValuePairs(string keyValuePairs, ILogger logger, bool shouldBeNonEmpty)
```

Comment:
This function is already implemented in `Microsoft.Azure.Devices.Edge.ModuleUtil.ModuleUtil`, so you can remove this impl and refer to that one.

Reply:
Ugh...I didn't realize `ModuleUtil` comes from the test tree. 🤕 Can you undo this particular change (restore the `ParseKeyValuePairs` method here, and remove the `ModuleUtils` reference from the csproj file). Sorry...

Reply:
Removed.